packages/openapi/src/decorators.ts

```diff
@@ -69,6 +69,10 @@ export const $extension: ExtensionDecorator = (
   setExtension(context.program, entity, extensionName as ExtensionKey, data);
 };
 
+export function setInfo(program: Program, entity: Namespace, data: AdditionalInfo) {
```

Comment:
```suggestion
export function setInfo(program: Program, entity: Namespace, data: AdditionalInfo & Record<ExtensionKey, unknown>) {
```

I think thats pretty good, wondering if you also should have `setInfoExtension` which adds to it instead of just replacing. But if thats good enough for you then this should be good

Reply:
For our particular use case, we will have validation against having both the `info` decorator and our custom decorator being defined together. And also, with the `info` object you can pass extensions directly on the decorator itself:

```
@info({"x-ms-test": "sample value"})
```

So I'd say this is good enough for our use case.